### PR TITLE
improvement-NXP-26126-rework-blob-enricher

### PR DESCRIPTION
**Pull request by ataillefer**
T&P: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-ataillefer/137/

### CODE REVIEW

**Comment by efge** [comment]
Please update [NXP-26126](https://jira.nuxeo.com/browse/NXP-26126) with an example of the resulting enriched blob, as documentation. I'm not sure why the upgrade notes of the ticket contain info about the added code though, it doesn't seem its place.

**Comment by efge** [comment]
👍

**Comment by nelsonsilva** [comment]
👍

File: nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/BlobAppLinksJsonEnricher.java [hunk 1]
```diff
@@ -32,21 +38,17 @@
...
-
-import static org.nuxeo.ecm.core.io.registry.reflect.Instantiations.SINGLETON;
-import static org.nuxeo.ecm.core.io.registry.reflect.Priorities.REFERENCE;
+import com.fasterxml.jackson.core.JsonGenerator;
 
 /**
  * Enrich {@link Blob} json with list of {@link AppLink}.
- *
+ * <p>
  * Enabled when parameter enrichers-blob=appLinks is present.
```
Comment (troger) [inline nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/BlobAppLinksJsonEnricher.java]: Not related directly to your PR but, don't we use `enrichers.blob=appLinks`?

**Comment by troger** [comment]
:+1:

File: nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/BlobAppLinksJsonEnricher.java (hunk 1 above)
Comment (ataillefer) [reply nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/BlobAppLinksJsonEnricher.java]: Agree, yet in most of the enricher's Javadoc this syntax is used...

Comment (nelsonsilva) [reply nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/BlobAppLinksJsonEnricher.java]: Not really, no, see https://jira.nuxeo.com/browse/NXP-18425

Comment (troger) [reply nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/BlobAppLinksJsonEnricher.java]: Right!

**Pull request merged**